dali/pipeline/data/dltensor.cc

```diff
@@ -19,10 +19,10 @@ namespace dali {
 
 DLDataType GetDLType(const TypeInfo &type) {
   DLDataType dl_type{};
-  DALI_TYPE_SWITCH(type.id(), T,
+  DALI_TYPE_SWITCH_WITH_FP16(type.id(), T,
       dl_type.bits = sizeof(T) * 8;
       dl_type.lanes = 1;
-      if (std::is_floating_point<T>::value) {
+      if (dali::is_fp_or_half<T>::value) {
```

Comment:
Bad idea. I'm not sure if fp16 is supported by dlpack at all. It supports bfloat but not fp16. Now you would put fp16 inside and claim it to be float.

Reply:
Check https://github.com/dmlc/dlpack/blob/main/include/dlpack/dlpack.h#L84

Reply:
Ok. Slice tests checks this indirectly.

Reply:
For the record: DLPack capsules created this way are compatible with PyTorch. My guess is that it's a _de-facto_ standard for float16 DLPack tensors now.